app/policies/application_policy.rb

```diff
@@ -71,6 +71,10 @@ def volunteer_related?
     volunteer? && record.volunteer.user_id == user.id
   end
 
+  def volunteer_included?
+    volunteer? && record.volunteers.include?(user.volunteer)
```

Comment:
They are of the same logic but the `record.user_ids.include?(user.id)` cannot be used here because the group assignments are associated with the volunteers (included the external ones, who will never become users) and not the users.